test/widgets/topic_list_test.dart

```diff
@@ -0,0 +1,375 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_checks/flutter_checks.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/initial_snapshot.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/route/channels.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/widgets/icons.dart';
+import 'package:zulip/widgets/message_list.dart';
+import 'package:zulip/widgets/page.dart';
+import 'package:zulip/widgets/topic_list.dart';
+import 'package:zulip/widgets/unread_count_badge.dart';
+
+import '../api/fake_api.dart';
+import '../model/binding.dart';
+import '../example_data.dart' as eg;
+import '../model/test_store.dart';
+import '../test_navigation.dart';
+import 'message_list_checks.dart';
+import 'page_checks.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  Widget? findRowByLabel(WidgetTester tester, String label) {
+    final textWidgets = tester.widgetList<Text>(find.text(label));
+    if (textWidgets.isEmpty) return null;
+
+    final rows = tester.widgetList<Row>(
+      find.descendant(
+        of: find.byType(TopicListItem),
+        matching: find.byType(Row),
+      ));
+
+    for (final row in rows) {
+      if (tester.widgetList(find.descendant(
+        of: find.byWidget(row),
+        matching: find.text(label),
+      )).isNotEmpty) {
+        return row;
+      }
+    }
+    return null;
+  }
```

Comment:
Given that we are always using `find.byWidget` on the result of this, how about just returning a finder?

```suggestion
  Finder findRowByLabel(WidgetTester tester, String label) {
    return find.descendant(
      of: find.byType(TopicListItem),
      matching: find.widgetWithText(Row, label));
  }
```